content/articles/django-celery-tasks/index.md

```diff
@@ -0,0 +1,449 @@
+
+# A page with the form where we can input the number of accounts to generate
+class GenerateRandomUserView(FormView):
+    template_name = 'task/generate_random_user.html'
+    form_class = GenerateRandomUserForm
+
+    def form_valid(self, form):
+        total = form.cleaned_data.get('total')
+        create_random_user_accounts.delay(total)
+        messages.success(self.request, 'We are generating your random users! Wait a moment and refresh this page.')
+        return redirect('users_list')
+```
+
+In the above code snippet, notice that we did not call the `create_random_user_accounts` method instead we called `create_random_user_accounts.delay(total)`. This instructs Celery to perform the task in a background process.
+
+#### URLs
+
+Update the `urls.py` with the code snippet below.
+
+```python
+from django.conf.urls import url
+from django.contrib import admin
+from django.urls import path
+
+from task.views import GenerateRandomUserView, UsersListView
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    url('users/', UsersListView.as_view(), name='users_list'),
+    url('generate/', GenerateRandomUserView.as_view(), name='generate')
+
+]
+```
+### Setting up base templates directory
+In the `settings.py` file, update the `TEMPLATES` section with the code snippet below.
+```python
+TEMPLATES = [
+    {
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'DIRS': [BASE_DIR / 'templates'], # We are adding in the directory where our templates will be stored.
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'context_processors': [
+                'django.template.context_processors.debug',
+                'django.template.context_processors.request',
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+            ],
+        },
+    },
+]
+
+``` 
+### Containerization
+#### Creating the Dockerfile
+In the root project directory, create a file named `Dockerfile` and add the code snippet below.
+```Dockerfile
+# pull the official base image
+FROM python:3.9.5-alpine
+
+# set work directory
+WORKDIR /usr/src/app
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# install dependencies
+RUN pip install --upgrade pip
+COPY ./requirements.txt /usr/src/app/requirements.txt
+RUN pip install -r requirements.txt
+
+# copy project
+COPY . /usr/src/app/
+
+RUN python manage.py migrate
+
+EXPOSE 8000
+
+CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]
+```
+The code snippet above contains directives that will be used to create the Docker image. For more information on how to create Django Docker images, visit [creating Django Docker images](https://www.section.io/engineering-education/django-docker/).
+
+#### Create a Docker compose deployment file
+In the working directory `project`, create a file named `docker-compose.yml` and add the code snippet below.
```

Comment:
You should add this file to project/celerytask. Please make it clear.

Reply:
The compose file should be added in the directory `project` not `project/celerytask`.